enum_methods.rb

```diff
@@ -0,0 +1,164 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    size.times do |i|
+      yield(to_a[i])
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    size.times do |i|
+      yield(to_a[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |i|
+      result << i if yield(i)
+    end
+    result
+  end
+
+  def my_all2?
+    return to_enum unless block_given?
+
+    my_each do |i|
+      return false unless yield(i)
+    end
+    true
+  end
+
+  def my_all?(arg = nil)
+    return false if !block_given? && arg.nil?
+
+    if arg.is_a?(Integer)
+      my_each do |i|
+        return false unless i == arg
+      end
+    elsif arg.is_a?(Regexp)
+      my_each do |i|
+        return false unless i.match(arg)
+      end
+    elsif !arg.is_a?(Integer) && !block_given?
+      my_each do |i|
+        return false unless i.is_a? arg
+      end
+    else
+      my_each do |i|
+        return false unless yield(i)
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    return false if !block_given? && arg.nil?
+
+    if arg.is_a?(Regexp)
+      my_each do |i|
+        return true if i.match(arg)
+      end
+    elsif arg.is_a?(Class)
+      my_each do |i|
+        return true if i.is_a? arg
+      end
+    elsif block_given?
+      my_each do |i|
+        return true if yield(i)
+      end
+    else
+      my_each do |i|
+        return true if i == arg
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    return false if !block_given? && arg.nil?
```

Comment:
* [ ] there should be a working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
Check the examples below and compare answers in each cases
```p %w{ant bear cat}.my_none?{ |word| word.length >= 4 }```!=```p %w{ant bear cat}.none?{ |word| word.length>=4 }```
```p [1, 3.14, 42].my_none?(Float)``` != ```p [1, 3.14, 42].none?(Float)```
```p [].my_none?``` !=```p [].none?```    
```p [nil].none?``` != ```p [nil].my_none?```
```p [nil, false].my_none?``` != ```p [nil, false].none?```